plutus-core/testlib/PlutusCore/Test.hs

```diff
@@ -340,6 +342,39 @@ goldenUEvalProfile name values =
   nestedGoldenVsDocM name ".ueval-profile" $
     pretty . view _2 <$> (rethrow $ runUPlcProfile values)
 
+-- Budget testing
+
+goldenUplcBudget ::
```

Comment:
I don't see the need to add anything new in this module. We already have `goldenBudget` and we can just use that.

Reply:
yes this is similar to `goldenBudget` but it's for `UPLC` not `PlutusTx`. It's less awkward to use because I can just input the `UPLC` programs. We have many similar `golden_` functions I don't think this is inconsistent with the code base.

Reply:
The goal is *not* to minimize the number of lines here. Doing unneeded conversions just to save 9 lines is just not worth it. On top of that it makes the code harder to read. Best to keep the code most direct. The code will be less bug prone too. Converting the arguments into `CompiledCode` obviously doesn't help here as it doesn't save any conversion.

Reply:
Even if we want to keep `goldenUplcBudget`, isn't this much better?

```haskell
goldenUplcBudget name term = goldenBudget name (DeserializedCode term Nothing mempty)
```

How is this harder to read or more bug prone?

Reply:
Since you discuss two options and can't agree which one is better, I'm going to do god's work and dump a third one on you: instead of defining `goldenBudget` in terms of `goldenUplcBudget`, do the opposite, because converting `CompiledCode` to `UPLC.Program` is something that `goldenBudget` already does anyway via `getPlcNoAnn`.

I don't see a reason to duplicate 10 lines of code here, particularly given that we have already had at least one case with a budget calculations not being consistent across different measurement functions. Also, `goldenBudget` is still `TallyingSt` while `goldenUplcBudget` is `CountingSt`?

Reply:
OK that direction of conversion does make more sense to me. I'll do that. 

I think I updated `goldenBudget` to be `CountingSt` already. Anyway it'll be combined anyway.

Reply:
> do the opposite

That seems sensible.

Reply:
Big win. I overall decrease the number of lines by 3 when adding `goldenUplcBudget`. Also the error messages are better, plus, there's no extra conversions at all. :)